TestFoundation/TestNSKeyedArchiver.swift

```diff
@@ -312,4 +313,25 @@ class TestNSKeyedArchiver : XCTestCase {
         let uuid = UUID()
         return test_archive(uuid, classes: [NSUUID.self])
     }
+
+    func test_archive_unhashable() {
+        let data = """
+            {
+              "args": {},
+              "headers": {
+                "Accept": "*/*",
+                "Accept-Encoding": "deflate, gzip",
+                "Accept-Language": "en",
+                "Connection": "close",
+                "Host": "httpbin.org",
+                "User-Agent": "TestFoundation (unknown version) curl/7.54.0"
+              },
+              "origin": "0.0.0.0",
+              "url": "https://httpbin.org/get"
+            }
+            """.data(using: .utf8)!
+        let json = try! JSONSerialization.jsonObject(with: data)
```

Comment:
I prefer to `do { }` this, then `XCTFail()` in the catch. That way a failure doesn't blow up the whole test suite.  What do you think?

Reply:
@ianpartridge I agree its definitely better to try and catch the error and fail the test then to prevent all of the other tests from even running.